### ISSUE TITLE
foreach loops not generating unit return value

### ISSUE THREAD

**Issue by cmester0** [hacspec/hacspec-v2#225]
Seems there are some issues with return values of foreach loops, as this
```
for zkp in zkps {
    check_valid(zkp);
}
```
results in the body
```
check_valid zkp
```
while
```
for zkp in zkps {
    check_valid(zkp); ()
}
```
generates
```
let _ := check_valid zkp in
tt
```
there should not be any difference returning unit or having `;` on the last line.

**Comment by W95Psp** [comment]
Hi @cmester0!
Yeah, that's not looking very nice...
In your example, `check_valid` is a function that returns `()`, right? (I guess it might panic or something?)

Though, I think deciding to drop that last `tt` or not is a bit difficult: the Rust compiler actually differentiates those.
The former is a block with one statement and no expression, the latter is a block with both a statement and a expression.

Thus, the ugly `let _ := ... in tt` is more faithful to what the user typed. But... yeah, it's ugly :sweat_smile: 

So idk what we should do.

**Comment by cmester0** [comment]
Yes, check_valid is a function returning `()` (just a dummy test for now).

My main issue is that the typing of the block changes, as it thinks the expression is the return value, while there should not be any return value because of `;`. I am not sure if we should distinguish between no return value and returning `()`. I can find many ways of returning `()`, so I think we should not try and remove the expression if present.

**Comment by W95Psp** [comment]
Wait, the type of `{check_valid(zkp); ()}` and `check_valid(zkp);` are different?

**Comment by cmester0** [comment]
Yes, sorry, the `check_valid` function actually returns a `bool`, so the two expressions are not the same type, as the block is interpreted as `check_valid()` not `check_valid();`.

**Comment by W95Psp** [comment]
(the bug it's probably the desugar of block in thir_import, if you wanna look at it @cmester0)

**Comment by W95Psp** [comment]
Was fixed by #228
